### PR TITLE
Add tests for CMSHomepage

### PR DESCRIPTION
**Pull request by gmrabian**
### Description

See title

About 300 of the line changes are yarn.lock. No need to be scared 😄 

---
### How to test

Tests pass

---
### Author checklist

- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
---

### CODE REVIEW

File: services/ui-src/package.json [hunk 1]
```diff
@@ -14,9 +14,10 @@
     "@okta/okta-react": "^4.1.0",
     "@reach/accordion": "^0.10.5",
     "@redux-devtools/extension": "^3.2.2",
+    "@testing-library/dom": "^9.3.3",
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.5.0",
-    "@testing-library/user-event": "^7.2.1",
+    "@testing-library/user-event": "^14.5.1",
```
Comment (gmrabian) [inline services/ui-src/package.json:20]: we saw a strange issue with `userEvent.click` where it was firing the `onClick` event twice. I noticed @benmartin-coforma had upgraded userevent in [another PR](https://github.com/Enterprise-CMCS/macpro-mdct-carts/pull/139504/files#diff-3709c9a794263a6b304bf1e61899bd231873695fc6931c550b57757bb3d8ad46), and some forum answers suggested that that might help, so I gave that a try. It did! Now the onClick event fires once per click and uses the correct parameters.

As a result, and per the [docs](https://testing-library.com/docs/user-event/intro/#writing-tests-with-userevent), I wrote these events like
```
const user = userEvent.setup();
...
await user.click(...)
```

File: services/ui-src/src/components/sections/homepage/CMSHomepage.js [hunk 2]
```diff
@@ -241,10 +241,10 @@ const CMSHomepage = ({
 CMSHomepage.propTypes = {
   getStatuses: PropTypes.func.isRequired,
   allStateStatuses: PropTypes.object.isRequired,
-  currentYear: PropTypes.object.isRequired,
+  currentYear: PropTypes.number.isRequired,
   currentUserRole: PropTypes.string.isRequired,
-  stateList: PropTypes.object.isRequired,
-  yearList: PropTypes.object.isRequired,
+  stateList: PropTypes.array.isRequired,
+  yearList: PropTypes.array.isRequired,
```
Comment (gmrabian) [inline services/ui-src/src/components/sections/homepage/CMSHomepage.js:247]: minor fixes in this file as a result of testing (yay!)

**Review by britt-mo — APPROVED** [review]
Tests passed for me (after running `yarn`)

**Review by benmartin-coforma — APPROVED** [review]
Very nice!

**Pull request merged**